### DISCUSSION TITLE
Add a button to reset the focal point

### DISCUSSION THREAD

**Discussion by TheBekker** [umbraco/Umbraco-CMS#10882] [Features and ideas]
With the introduction of Media Picker v3 we can now also create focal points locally on the picker, but we still don't have a way to reset/remove that focal point.

Thinking it could be a simple button or even double clicking the focal point, to reset it.

**Comment by nul800sebastiaan** [comment]
We've never had a way to reset the focal point before either! 😉

Does it make sense to reset it to the middle though? I've you've changed and saved it, it seems like that would have been intentional. And if you want to change it, as an editor, are you concerned that the focal point is not exactly in the middle? Wouldn't just just want to improve the location?

**Reply by TheBekker** [reply]
That was also my though to begin with, but it suddenly became an issue in the project i'm on right now. :)

Our exact use case is that we have pages that contains a lot of images of kitchens, that are framed similarly.
So if an editor accidently or even intentional changes the focal point, but then regrets, its very obvious that the image is not cropped at the exact same location as the rest, because it's hard to get in perfectly in the middle again. (if that makes sense)

But beside that, It also feel like it's a normal thing, being able to undo a change you make.

**Reply by nul800sebastiaan** [reply]
Yeah not saying it's a bad idea, just throwing some thoughts out there! 

Can I then ask: why are they allowed to change the focal point at all if they shouldn't? 😅

**Reply by TheBekker** [reply]
Yeah i understand :)

Well they should be able to, they should just also be able to change their mind about it :)

**Reply by nul800sebastiaan** [reply]
I'll bring it up with the frontend team and see what they say.

**Reply by TheBekker** [reply]
And i'll be happy to see if i can't make time to do a PR with a suggestion in the weekend if it helps.

**Reply by nul800sebastiaan** [reply]
I'll make sure to send you some encouragement based on what they have to say about it :)

**Reply by nul800sebastiaan** [reply]
@TheBekker Sending encouragement! 

From my colleague: 

> The current convention in the backoffice is to enable the button when the focal point is changed and disable it when you reset it. Then the action is always visible.

If you're up for it, we'd love to see a PR. :) 

And for bonus points he says:

> We could also implement the same logic to the reset crop button.

**Reply by TheBekker** [reply]
Awesome, i'll see if i can't whip something up in the weekend.
I'm going to get my 2nd shot, so hopefully that won't take me out :)

**Comment by ronaldbarendse** [comment]
This would make absolute sense, as the codebase behaves quite differently when the focal point isn't exactly in the middle:
https://github.com/umbraco/Umbraco-CMS/blob/2980ff9c54fe7c4bdf4f0c19a5672f148c3cd5d2/src/Umbraco.Core/PropertyEditors/ValueConverters/ImageCropperValue.cs#L136-L137

https://github.com/umbraco/Umbraco-CMS/blob/2980ff9c54fe7c4bdf4f0c19a5672f148c3cd5d2/src/Umbraco.Core/PropertyEditors/ValueConverters/ImageCropperValue.cs#L66-L75

-----------

Just a minor detail in regards to Media Picker v3: when the local focal point is disabled, the code ensures the local focal point is set to `null` and when merging local crops with the media crops, it will therefore always use the media item one:

https://github.com/umbraco/Umbraco-CMS/blob/2980ff9c54fe7c4bdf4f0c19a5672f148c3cd5d2/src/Umbraco.Web/PropertyEditors/MediaPicker3Configuration.cs#L94-L97

https://github.com/umbraco/Umbraco-CMS/blob/2980ff9c54fe7c4bdf4f0c19a5672f148c3cd5d2/src/Umbraco.Core/PropertyEditors/ValueConverters/ImageCropperValue.cs#L178

But when enabling the local focal point while there is already picked media stored without a focal point, we should ensure it defaults to the middle and doesn't fallback to the media items focal point. Otherwise just opening the picked media and resaving the content would change the focal point from the media item one to the picked media one (I hope this all makes sense) 😃

**Comment by TheBekker** [comment]
Made PR #10923 .
Since it contains new text, i'm not sure what the requirement are in that regards.

I've added translation for danish and english.
Let me know what you think of this simple suggestion.

**Comment by nul800sebastiaan** [comment]
This is now merged and will be available in 8.17.0 https://github.com/umbraco/Umbraco-CMS/pull/10923 👍
